go/chat/sender.go

```diff
@@ -85,28 +91,41 @@ func (s *BlockingSender) addPrevPointersAndCheckConvID(ctx context.Context, msg
 		return chat1.MessagePlaintext{}, fmt.Errorf("addPrevPointersToMessage expects an empty prev list")
 	}
 
-	var prevs []chat1.MessagePreviousPointer
-
 	res, err := s.G().ConvSource.Pull(ctx, conv.GetConvID(), msg.ClientHeader.Sender,
 		&chat1.GetThreadQuery{
 			DisableResolveSupersedes: true,
 		},
-		&chat1.Pagination{
-			Num: 50,
-		})
+		s.prevPtrPagination)
 	if err != nil {
 		return chat1.MessagePlaintext{}, err
 	}
 
 	if len(res.Messages) == 0 {
 		s.Debug(ctx, "no local messages found for prev pointers")
 	}
-	prevs, _, err = CheckPrevPointersAndGetUnpreved(&res)
+	var prevs []chat1.MessagePreviousPointer
+	newPrevsForRegular, newPrevsForExploding, err := CheckPrevPointersAndGetUnpreved(&res)
 	if err != nil {
 		return chat1.MessagePlaintext{}, err
 	}
 
-	if len(prevs) == 0 {
+	var numPrev int
+	if msg.IsEphemeral() {
+		prevs = newPrevsForExploding
+		numPrev = len(newPrevsForExploding)
+	} else {
+		prevs = newPrevsForRegular
+		// If we have only sent ephemeralMessages and are now sending a regular
+		// message, we may have an empty list for newPrevsForRegular. In this
+		// case we allow the `Prev` to be empty, so we don't want to abort in
+		// the check on numPrev below.
+		numPrev = len(newPrevsForRegular)
+		if numPrev == 0 {
+			numPrev = len(newPrevsForExploding)
```

Comment:
This line is kind of setting `numPrev` to the wrong value, because it knows all that matters is whether the value is zero or not. Maybe it would be cleaner to call is `hasPrevs` or something, and just make it a bool?

Reply:
yeah i like that i'll switch